cmd/krew/cmd/root.go

```diff
@@ -150,13 +150,8 @@ func preRun(cmd *cobra.Command, _ []string) error {
 	}
 
 	if _, ok := os.LookupEnv(constants.EnableMultiIndexSwitch); ok {
-		isMigrated, err = indexmigration.Done(paths)
-		if err != nil {
-			return errors.Wrap(err, "error getting file info")
-		}
-		if !isMigrated && cmd.Use != "index-upgrade" {
-			fmt.Fprintln(os.Stderr, "You need to perform a migration to continue using krew.\nPlease run `kubectl krew system index-upgrade`")
-			return errors.New("krew index outdated")
+		if err := indexmigration.Migrate(paths); err != nil {
+			return err
```

Comment:
this err needs to be wrapped. I don't see any indication of what was being done for err messages inside Migrate() (saying "migration failed"). So it's good to add that here.

Reply:
I don't think we should say "migration failed", because the index could already have been migrated, but the check failed. Wrapping with a message which communicates this ambiguity would be fine though.

Maybe the function is also not named well. It checks and maybe migrates instead of just migrate.

Reply:
What do you guys think about moving the migration needed check out of `Migrate` then? I think that would solve these issues.

Reply:
Yes, let's separate the check from the migration. That will make it easier to understand what is actually happening. :+1: